src/MongoDB.Driver.Core/Core/Servers/TopologyDescription.cs

```diff
@@ -0,0 +1,323 @@
+/* Copyright 2020–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using MongoDB.Bson;
+using MongoDB.Shared;
+#if NET452 || NETSTANDARD2_0
+using System.Runtime.Serialization;
+#endif
+
+namespace MongoDB.Driver.Core.Servers
+{
+    /// <summary>
+    /// Represents a topology description.
+    /// Comparing topology descriptions freshness does not exhibit the reversal property of
+    /// inequalities e.g. a.IsStalerThan(b) (a "&lt;" b) does not imply !b.IsStalerThan(a) (b "&gt;" a)
+    /// See <seealso cref="CompareFreshnessToServerResponse"/> for more information.
+    /// </summary>
+#if NET452 || NETSTANDARD2_0
+    [Serializable]
+    public readonly struct TopologyDescription : IEquatable<TopologyDescription>, ISerializable
```

Comment:
See ServerId for a similar `#if`

Reply:
I created this structure as a class, but I guess a struct is also fine here.
NOTE: `public readonly` is interesting feature

Reply:
I wasn't sure about the name, but I called variables and the type definition in the same way `TopologyVersion`. Not sure which name is better, but at least I would say that they should be the same?

Reply:
I think your name is way better :) I've changed mine!